File: client-api/src/test/java/org/ostelco/topup/api/util/AccessToken.java
package org.ostelco.topup.api.util;

import io.jsonwebtoken.Jwts;
import io.jsonwebtoken.SignatureAlgorithm;
import io.vavr.collection.HashMap;
import java.util.List;
import io.vavr.collection.Array;

import java.util.Map;

public class AccessToken {

    private final static String key = "secret";
    private final static String namespace = "https://ostelco.org";

    private AccessToken() {
    }

    public static String withEmail(final String email) {

        List<String> audience = Array.of("http://kmmtest", namespace + "/userinfo")
            .toJavaList();

        return AccessToken.withEmail(email, audience);
    }

    public static String withEmail(final String email, List<String> audience) {

        final Map<String, Object> claims = HashMap
                .of(namespace + "/email", (Object) email,
                    "aud", (Object) audience,
                    "sub", (Object) email)
                .toJavaMap();

        return Jwts.builder()
                .setClaims(claims)
                .signWith(SignatureAlgorithm.HS512, key)
                .compact();
    }
}

File: client-api/src/test/kotlin/org/ostelco/prime/client/api/util/AccessToken.kt
package org.ostelco.prime.client.api.util

import io.jsonwebtoken.Jwts
import io.jsonwebtoken.SignatureAlgorithm
import io.vavr.collection.Array
import io.vavr.collection.HashMap

object AccessToken {

    private val key = "secret"
    private val namespace = "https://ostelco.org"

    fun withEmail(email: String): String {

        val audience = Array.of("http://kmmtest", "$namespace/userinfo")
                .toJavaList()

        return withEmail(email, audience)
    }

    fun withEmail(email: String, audience: List<String>): String {

        val claims = HashMap
                .of("$namespace/email", email as Any,
                        "aud", audience as Any,
                        "sub", email as Any)
                .toJavaMap()

        return Jwts.builder()
                .setClaims(claims)
                .signWith(SignatureAlgorithm.HS512, key)
                .compact()
    }
}
